common/include/opae/cxx/core/except.h

```diff
@@ -99,10 +111,259 @@ class except : public std::exception {
+ * for exceptions thrown when the error code
+ * FPGA_NO_ACCESS is returned from a call to
+ * an OPAE C API function
+ */
+class no_access : public except {
+  public:
+
+    /** no_access constructor
+     *
+     * @param[in] loc Location where the exception was constructed.
+     */
+    no_access(src_location loc) noexcept
+    : except(FPGA_NO_ACCESS, "failed with return code FPGA_NO_ACCESS", loc)
+    {}
+
+};
+
+/** reconf_error exception
+ *
+ * reconf_error tracks the source line of origin
+ * for exceptions thrown when the error code
+ * FPGA_RECONF_ERROR is returned from a call to
+ * an OPAE C API function
+ */
+class reconf_error : public except {
+  public:
+
+    /** reconf_error constructor
+     *
+     * @param[in] loc Location where the exception was constructed.
+     */
+    reconf_error(src_location loc) noexcept
+    : except(FPGA_RECONF_ERROR, "failed with return code FPGA_RECONF_ERROR", loc)
+    {}
+
+};
+
+namespace detail {
+
+/** typedef function pointer that returns bool if result is FPGA_OK
+ */
+typedef bool (*exception_fn)(fpga_result, const opae::fpga::types::src_location & loc);
+
+/** is_ok is a template function that throws an excpetion of its template argument type
+ *  if the result code is not FPGA_OK. Otherwise it returns true.
+ */
+template<typename T>
+constexpr bool is_ok(fpga_result result, const opae::fpga::types::src_location & loc) {
+  return result == FPGA_OK ? true : throw T(loc);
+}
+
+
+static exception_fn opae_exceptions[12] = { is_ok<opae::fpga::types::invalid_param>,
+                                            is_ok<opae::fpga::types::busy>,
+                                            is_ok<opae::fpga::types::exception>,
+                                            is_ok<opae::fpga::types::not_found>,
+                                            is_ok<opae::fpga::types::no_memory>,
+                                            is_ok<opae::fpga::types::not_supported>,
+                                            is_ok<opae::fpga::types::no_driver>,
+                                            is_ok<opae::fpga::types::no_daemon>,
+                                            is_ok<opae::fpga::types::no_access>,
+                                            is_ok<opae::fpga::types::reconf_error> };
+
+
+static inline void assert_fpga_ok(fpga_result result,
+                                  const opae::fpga::types::src_location & loc) {
+  if (result > FPGA_OK && result <= FPGA_RECONF_ERROR)
+    // our exception table above starts at invalid_param with index 0
+    // but FPGA_INVALID_PARAM is actually enum 1 - let's account for that
+    opae_exceptions[result-1](result, loc);
+}
+
+}  // end of namespace detail
+
+/// Macro to check of result is FPGA_OK
+/// If not, throw exception that corresponds
+/// to the result code
+#define ASSERT_FPGA_OK(r) \
+  opae::fpga::types::detail::assert_fpga_ok(r, \
+      opae::fpga::types::src_location(__FILE__, __func__, __LINE__));
```

Comment:
Looks like you got rid of the code that shortens __FILE__ to the filename.  This will display whatever relative path the compiler happens to use.  Not sure we wan't that.

Reply:
Nvm, I see it now.